File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.cpp
class Solution {
public:
    int pivotInteger(int n) {
        if(n==1) return 1;
        int j=1,res=0,ans;
        for(int i=n;i>=1;i--) {   
            res+=i;
           ans=n-j;
            cout<<res<<" " <<ans*(ans+1)/2<<endl;
            if(ans*(ans+1)/2==res+ans ) { return ans; } 
         j++;
            
        }
    return -1;
        
    }
};

File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.js
/**
 * @param {number} n
 * @return {number}
 */
var pivotInteger = function(n) {
      if(n==1) return 1;
        let j=1,res=0,ans;
        for(let i=n;i>=1;i--) {   
            res+=i;
           ans=n-j;
            if(ans*(ans+1)/2==res+ans ) { return ans; } 
         j++;
            
        }
    return -1;
};